### PR TITLE
[FIX] website_sale_delivery: check carrier before payment

### PR DESCRIPTION
**Pull request by Dj0ulo**
Check if the user selected the carrier for storable/consumable products before proceeding to payment.

Backport of [35ea9bf](https://github.com/odoo/odoo/commit/35ea9bfb)

opw-3861697

### CODE REVIEW

File: addons/website_sale_delivery/controllers/main.py [hunk 1]
```diff
@@ -133,3 +133,13 @@ def _update_website_sale_delivery_return(self, order, **post):
                 'new_amount_total_raw': order.amount_total,
             }
         return {}
+
+
+class PaymentPortal(PaymentPortal):
```
Comment (Feyensv) [inline addons/website_sale_delivery/controllers/main.py:138]: > class already defined line 6 (E0102)

**Review by Feyensv — COMMENTED** [review]
@vchu-odoo To be sure, do you remember why we didn't fix it in earlier versions?
I suppose it's because of the changes in `website_sale_delivery` in 16.2/16.3, to avoid conflicts ?

**Review by Feyensv — APPROVED** [review]
@robodoo delegate+ up to 16.0

**Comment by Dj0ulo** [comment]
@robodoo r+

**Comment by Dj0ulo** [comment]
@robodoo r+

**Comment by Dj0ulo** [comment]
@robodoo retry

**Pull request closed without merging**